reana_commons/api_client.py

```diff
@@ -0,0 +1,116 @@
+# A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# REANA; if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+# Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+"""REANA REST API base client."""
+
+import json
+import os
+
+import pkg_resources
+from bravado.client import SwaggerClient
+
+
+class BaseAPIClient(object):
+    """REANA API client code."""
+
+    def __init__(self, component, server_data, http_client=None):
+        """Create a OpenAPI client for a REANA component."""
+        self.component = component
+        server_url, spec_file = server_data
+        json_spec = self._get_spec(spec_file)
+        self._client = SwaggerClient.from_spec(
+            json_spec,
+            http_client=http_client,
+            config={'also_return_response': True})
+        self._client.swagger_spec.api_url = server_url
+        self.server_url = server_url
+
+    def _get_spec(self, spec_file):
+        """Get json specification from package data."""
+        spec_file_path = os.path.join(
+            pkg_resources.
+            resource_filename(
+                'reana_commons',
+                'openapi_connections'),
+            spec_file)
+
+        with open(spec_file_path) as f:
+            json_spec = json.load(f)
+        return json_spec
+
+
+class JobControllerAPIClient(BaseAPIClient):
+    """REANA-Job-Controller http client class."""
+
+    def submit(self,
+               experiment='',
+               image='',
+               cmd='',
+               prettified_cmd='',
+               workflow_workspace='',
+               job_name=''):
+        """Submit a job to RJC API.
+
+        :param name: Name of the job.
+        :param experiment: Experiment the job belongs to.
+        :param image: Identifier of the Docker image which will run the job.
+        :param cmd: String which represents the command to execute. It can be
+            modified by the workflow engine i.e. prepending ``cd /some/dir/``.
+        :prettified_cmd: Original command submitted by the user.
+        :return: Returns a dict with the ``job_id``.
+        """
+        job_spec = {
+            'experiment': experiment,
+            'docker_img': image,
+            'cmd': cmd,
+            'prettified_cmd': prettified_cmd,
+            'env_vars': {},
+            'workflow_workspace': workflow_workspace,
+            'job_name': job_name,
+        }
+
+        response, http_response = self._client.jobs.create_job(job=job_spec).\
+            result()
+        return response
+
+    def check_status(self, job_id):
+        """Check status of a job."""
+        response, http_response = self._client.jobs.get_job(job_id=job_id).\
```

Comment:
Should we check, here in `submit` and in `get_logs`, the `http_response.status_code` and raise an exception in case it is different than `200`? Also `ConnectionError` can happen like in `check_if_cached`.

Reply:
I agree, this is still a todo